pkg/maps/ctmap/ctmap.go

```diff
@@ -360,24 +363,38 @@ func (f *GCFilter) doFiltering(dstIP net.IP, dstPort uint16, nextHdr, flags uint
 		action = deleteEntry
 	}
 
-	// used by FlushCTEntriesOf
-	// Delete all entries of the given dstIP that are not filtered
-	if f.Type&GCFilterByIDsToKeep != 0 &&
-		dstIP.Equal(f.IP) &&
-		// only delete TUPLE_F_IN connections since the clean up is made
-		// for ingress policies
-		flags&TUPLE_F_IN != 0 {
+	// If the filter doesn't contain an endpoint ID & IP, no entries will get matched below.
+	if f.EndpointID == 0 || f.EndpointIP == nil {
+		return
+	}
+
+	// Determine whether the entry matches the endpoint IP,
+	// and the direction of the entry (ingress or egress).
+	var ingress bool
+	if flags&TUPLE_F_IN != 0 && dstIP.Equal(f.EndpointIP) {
+		ingress = true
+	} else if flags&TUPLE_F_IN == 0 && srcIP.Equal(f.EndpointIP) {
```

Comment:
@tgraf @joestringer can you please confirm that this is correct, ie that we store the source address of egress flows into the daddr field in the conntrack entry key?

Reply:
Yes.

The addresses correspond to the expected addresses of the flow to be allowed.

The flag specifies the direction in which the entry was created, if an ingress packet triggered the entry, it will have flag `TUPLE_F_IN`